qiita_db/support_files/patches/python_patches/47.py

```diff
@@ -0,0 +1,30 @@
+from qiita_db.study import Study
+
+
+class ForRecursion(object):
```

Comment:
I think you are facing the same issue that I was facing when doing the analysis refactor. Try changing this:

https://github.com/biocore/qiita/pull/2040/files#diff-694f6ed1529987bc3435746a74426405R408

then the functions should work